packet/io.go

```diff
@@ -36,29 +36,27 @@ import (
 // found. It returns the offset of the sync w.r.t. the original reader
 // position.
 func Sync(r *bufio.Reader) (int64, error) {
-	data := make([]byte, 1)
 	for i := int64(0); ; i++ {
-		_, err := io.ReadFull(r, data)
+		data, err := r.Peek(1)
 		if err == io.EOF || err == io.ErrUnexpectedEOF {
 			break
 		}
 		if err != nil {
 			return 0, err
 		}
 		if int(data[0]) == SyncByte {
-			// check next 188th byte
-			rp := bufio.NewReaderSize(r, PacketSize) // extends only if needed
-			nextData, err := rp.Peek(PacketSize)
+			// Make sure there is a full TS packet
+			_, err := r.Peek(PacketSize)
 			if err == io.EOF || err == io.ErrUnexpectedEOF {
 				break
 			}
 			if err != nil {
 				return 0, err
 			}
-			if nextData[187] == SyncByte {
```

Comment:
I don't recall the exact reason but I believe there was a reason to check both sync bytes before advancing and returning the new position. We are no longer doing that. (At the least we are at odds with the docstring.)

Reply:
Upstream version of this method does not check for the second sync byte anymore. It seems safe to make this change. I have updated the docstring.